src/commands/rankup.ts

```diff
@@ -143,56 +135,102 @@ async function getNextRank(
+  .addStringOption((option) =>
+    option
+      .setName("username")
+      .setDescription("The Codewars username to query rankup statistics for")
+  )
+  .addStringOption((option) =>
+    option.setName("target").setDescription("The target user or rank to reach")
+  )
+  .addStringOption((option) =>
+    option
+      .setName("language")
+      .setDescription("The programming language ID to query rankup statistics for")
+  )
+  .addStringOption((option) =>
+    option
+      .setName("mode")
+      .setDescription("Choose your preferred method to reach your target")
+      .addChoices(modes.map((mode) => [mode.description, mode.name]))
+  )
+  .addStringOption((option) =>
+    option
+      .setName("limit")
+      .setDescription("Don't suggest Kata above this rank")
+      .addChoices(limits.map((limit) => [limit, limit]))
+  )
+  .addBooleanOption((option) =>
+    option.setName("ephemeral").setDescription("Don't show rank up statistics to others")
+  )
+  .toJSON();
+
+export const call = async (interaction: CommandInteraction) => {
+  let username = interaction.options.getString("username");
+  if (!username) {
+    const member = interaction.member;
+    const displayName = member instanceof GuildMember ? member.displayName : member?.nick;
+    if (!displayName) {
+      await interaction.reply({
+        content: "Failed to fetch the name of the current user",
+        ephemeral: true,
+      });
       return;
     }
-    [username] = result.data;
+    username = displayName;
+  }
+  const target = interaction.options.getString("target");
+  const language = interaction.options.getString("language");
+  const mode = interaction.options.getString("mode") || DEFAULTMODE;
+  const limit = interaction.options.getString("limit") || "1kyu";
+  const ephemeral = interaction.options.getBoolean("ephemeral") || false;
+
+  // Restrict command output to #bot-playground unless ephemeral is set
+  if (!ephemeral && (interaction.channel as TextChannel).name !== "bot-playground") {
+    await interaction.reply({
+      content: REDIRECT,
+      ephemeral: true,
+    });
+    return;
   }
-
-  // If help option set, simply reply with usage info
-  if (opts.help !== undefined) return send(USAGE);
-
-  // Get mode
-  const mode = isMode(opts.mode) ? opts.mode : DEFAULTMODE;
-
-  // Validate limit
-  if (opts.limit && !/^[1-8]$/.test(opts.limit)) return send("Invalid limit");
 
   // Get user data
   let score: number;
   try {
-    score = await getUser(username, opts.language);
+    score = await getUser(username, language);
   } catch (err) {
-    return send(errorMessage(err));
+    await interaction.reply({
+      content: errorMessage(err),
+      ephemeral: true,
+    });
+    return;
+  }
+  if (score == 0) {
+    await interaction.reply({
+      content: `${username} has not started training \`${language}\`, or the language ID is invalid.
+See <https://docs.codewars.com/languages/> to find the correct ID.`,
+      ephemeral,
```

Comment:
I don't think this should be visible either.
```suggestion
      ephemeral: true,
```